pandas/core/generic.py

```diff
@@ -7132,28 +7132,21 @@ def _clip_with_scalar(self, lower, upper, inplace=False):
                 (upper is not None and np.any(isna(upper)))):
             raise ValueError("Cannot use an NA value as a clip threshold")
 
-        result = self.values
-        mask = isna(result)
-
-        with np.errstate(all='ignore'):
-            if upper is not None:
-                result = np.where(result >= upper, upper, result)
-            if lower is not None:
-                result = np.where(result <= lower, lower, result)
-        if np.any(mask):
-            result[mask] = np.nan
+        result = self
 
-        axes_dict = self._construct_axes_dict()
-        result = self._constructor(result, **axes_dict).__finalize__(self)
+        if upper is not None:
+            subset = self.le(upper, axis=None) | isna(result)
+            result = result.where(subset, upper, axis=None, inplace=inplace)
+        if lower is not None:
+            if inplace:
```

Comment:
Why is this `inplace` check only when `lower is not None`? It looks like you also assign `result = self` before the upper and lower check.

Reply:
Got it. Thanks!

Reply:
I think its just better to use the existing idiom

```
if inplace:
     self._update_inplace(result0
else:
    return self
```

Reply:
actioned. thanks.